File: 326. Power of Three/main.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        val = 1

        while val < n:
            val *= 3

        if val == n:
            return True

        return False


class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n == 1:
            return True

        if n < 1 or n == 2:
            return False

        if n % 3 == 0:
            return self.isPowerOfThree(n / 3)

        return False


class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n == 1:
            return True

        if n < 1 or n % 3 != 0:
            return False

        return self.isPowerOfThree(n / 3)


class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        return n > 0 and 1162261467 % n == 0

File: 326. Power of Three/main.rs
struct Solution;

impl Solution {
    pub fn is_power_of_three(n: i32) -> bool {
        let mut val: i64 = 1;

        while val < n as i64 {
            val *= 3;
        }

        if (val == n as i64) {
            return true;
        }

        return false;
    }
}

impl Solution {
    pub fn is_power_of_three(n: i32) -> bool {
        if n == 1 {
            return true;
        }

        if n < 1 || n == 2 {
            return false;
        }

        if n % 3 == 0 {
            return Self::is_power_of_three(n / 3);
        }

        return false;
    }
}

impl Solution {
    pub fn is_power_of_three(n: i32) -> bool {
        if n == 1 {
            return true;
        }

        if n < 1 || n % 3 != 0 {
            return false;
        }

        return Self::is_power_of_three(n / 3);
    }
}


impl Solution {
    pub fn is_power_of_three(n: i32) -> bool {
        return n > 0 && 1162261467 % n == 0;
    }
}